frontend/public/components/cluster-settings/global-config.tsx

```diff
@@ -115,21 +115,24 @@ const GlobalConfigPage_: React.FC<GlobalConfigPageProps & GlobalConfigPageExtens
         };
       });
       const allItems = [...winnowedResponses, ...usableConfigs]
-        .map((item) => {
-          const apiExplorerLink = `/api-resource/cluster/${referenceForModel(item.model)}`;
-          const resourceLink = resourcePathFromModel(item.model, item.name, item.namespace);
-          return {
-            label: item.model.kind,
-            apiGroup: item.model.apiGroup,
-            id: item.id,
-            description: getResourceDescription(item.model),
-            path: resourceLink,
-            menuItems: [
-              editYAMLMenuItem(item.model.kind, resourceLink),
-              viewAPIExplorerMenuItem(item.model.kind, apiExplorerLink),
-              ...(item.model.kind === 'OAuth' ? oauthMenuItems : []),
-            ],
-          };
+        .flatMap((item) => {
```

Comment:
Why do we need to flatten the returned array ?

Reply:
Some items in the original array are undefined, so returning the flattened array filters them out. Otherwise we will render some elements in the console that are blank